sympy/functions/elementary/exponential.py

```diff
@@ -495,8 +495,13 @@ def _eval_nseries(self, x, n, logx, cdir=0):
             return Order(x**n, x)
         if arg0 is S.Infinity:
             return self
+        for arg1 in arg.args:
+            if isinstance(arg1, log):
+                term = arg/arg1
+                if not (term).has(x) and exp(arg1).has(x) and term.is_number:
+                    return self
```

Comment:
I would say this is a bit too complex for what I would like to have. Consider the following
1) We have the tools required to compute something like `(exp(-I*pi*(2*x+1)))._eval_nseries(x, 3 , None)` but currently we are returning `self` for this. So we don't need to worry about computing this.
2) Consider something like this `exp(I*log(x))._eval_nseries(x, 3 , None)`
If you see we already are handling cases where `arg0` is `oo/-oo`
```
        if arg0 is S.NegativeInfinity:
            return Order(x**n, x)
        if arg0 is S.Infinity:
            return self
```
What we are not handling are cases where `arg0` is `infinite` but not `oo/-oo` (something like `I*oo`) , and our example above falls in this category.
Also if you think about if we don't really expect any series expansion for cases like `exp(I*log(x))`, what we can expect is a `PoleError`. Here's how we could approach this
```
>>> exp(I*log(x))
exp(I*log(x))
>>> _.rewrite(sin)
I*sin(log(x)) + cos(log(x))
>>> _.series(x)
Traceback (most recent call last):
sympy.core.function.PoleError: Cannot expand sin(log(_x)) around 0
>>> sin(log(x)).series(x)
Traceback (most recent call last):
sympy.core.function.PoleError: Cannot expand sin(log(_x)) around 0
>>> cos(log(x)).series(x)
Traceback (most recent call last):
sympy.core.function.PoleError: Cannot expand cos(log(_x)) around 0
```
Hence we could address all of these points above.

Reply:
Yeah, raising a PoleError for cases that are trivially known to have no expansion is a better way.